manager/controllers/app/m4dapplication_controller.go

```diff
@@ -374,6 +373,7 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	blueprintPerClusterMap := r.GenerateBlueprints(instances, applicationContext)
 	setReadModulesEndpoints(applicationContext, blueprintPerClusterMap, moduleMap)
 	resourceRef := r.ResourceInterface.CreateResourceReference(applicationContext.Name, applicationContext.Namespace)
+	resourceRef.AppVersion = applicationContext.GetGeneration()
```

Comment:
maybe this should be part of the `CreateResourceReference` function if it's part of the same struct and not set separately.

Reply:
ok, will change.